### DISCUSSION TITLE
Models are undefined

### DISCUSSION THREAD

**Discussion by hieunc229** [prisma/prisma#20209] [Q&A]
### Bug description

I created a schema, generate the client (with typescript). But it shows models are undefined

### How to reproduce

1. Create schema (using a custom path, ie `projectRoot/prisma/custom.prisma`)
2. Run `npx prisma generate --schema=./prisma/custom.prisma`
3. Run `npx prisma migrate dev --name init --schema=./prisma/custom.prisma`
4. Now, the new PrismaClient is created
5. Create prisma instance (import PrismaClient as instruction) and start express server
6. When query/create/update a record (ie `prisma.user.findMany`, or `prisma.user.create`), it shows error `Cannot read properties of undefined (reading 'findMany')`

Using typescript 5.0.4, prisma 5.0.0 

### Prisma information

custom.prisma
```prisma
generator client {
  provider = "prisma-client-js"
  output   = env("PRISMA_OUTPUT")
}

datasource db {
  provider = "sqlite"
  url      = env("PRISMA_DATABASE_URL")
}

model User {
  id         String   @id
  email      String   @unique
  name       String?
  created    BigInt
}
```

index.ts
```ts
import { PrismaClient } from "../prisma-client/orders";
const prisma = new PrismaClient()
prisma.user
          .create({
            data: {
              id: "userid",
              created: new Date().getTime()
            },
          })
```

### Environment & setup

- OS: masOS
- Database: Sqlite
- Node.js version: v18.16.0

### Prisma Version

```
5.0.0
```

**Comment by janpio** [comment]
Is this a new problem with Prisma 5 or are you a new Prisma user?

The steps you describe sound valid. Does the CLI output of `prisma generate` and `migrate dev` show the correct schema path?

Are you sure that `output   = env("PRISMA_OUTPUT")` and `from "../prisma-client/orders"` are the same path? Maybe delete that folder once, and see if it is successfully recreated on another run of `prisma generate`.

(By the way, you can also configure the path of your Prisma schema file in your `package.json` file, then you do not have to use `--schema` for each command: https://www.prisma.io/docs/concepts/components/prisma-schema#prisma-schema-file-location)

**Reply by hieunc229** [reply]
I have been using Prisma before (but not often). This is a new project and I found it strange too. I tried to install version `4.16` and the same thing happened

Yes, I'm sure the `output = env("PRISMA_OUTPUT")` is `../prisma-client/orders"...?`. I tried to remove `prismat-client`, `migrations`, `node_modules`, and regenerate the files again. The issue persist

**Reply by janpio** [reply]
Then the question is: Why is the generated client missing those models that you define in your Schema file?

The generated client includes a copy of `schema.prisma` - can you check that what it looks like? If it is not the same as your schema, something is off with the paths and your `prisma generate` is writing the file to another location from that you are importing.

**Reply by hieunc229** [reply]
Yes, it looks exactly as the original prisma file. The generated type files looks good as well (the models are included)

**Reply by janpio** [reply]
So the types are fine, but executing the code you then get an error? Can you shared the full error you are actually getting?
Anything special about your project?

**Reply by hieunc229** [reply]
My project is built in Typescript, with expressjs. The only thing I do differently (that I notice) is using the custom schema path. Other than that, I don't think there is anything special

Sorry I've removed prisma for now, and didn't save the full error. It showed something like

```
Cannot read properties of undefined (reading 'findMany')
   userGetInfo.ts line 18
   ....
```

Where `userGetInfo.ts` is where I use `prisma.user.findMany`

I think it's quite unusual as I've use it before. Thanks for your support and wish I could provide more details

**Reply by janpio** [reply]
Let me know if you try again, or figure out what was going on.

**Reply by faisal-aali** [reply]
Hi @janpio not OP but I'm facing the same error.

`Cannot read properties of undefined (reading 'findFirst')`

here

```ts
 const userExists = await prisma.user.findFirst({
    where: { login: user.login },
  });
```
_`prisma` is also being logged as `{}` here_

even tho I do have that code generated in client

` get user(): Prisma.UserDelegate<ExtArgs>`

**Comment by M-krishna** [comment]
Any update on this issue? Has anybody found a solution for this?

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that haven’t had any recent activity and appear to be outdated.

If this discussion is still important to you or unresolved, we’d love to hear from you! Feel free to reopen it or start a new one with updated details.

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and being part of the community!
